### PR TITLE
Issue 20153 rules showing incomplete components and order

### PR DESCRIPTION
**Pull request by alfredo-dotcms**
### Proposed Changes
* a condition value for a session attribute is not possible
* do not deactivate the rule unless the rule is invalid
* browser conditionlet is out of order
* Logger must be deactivated in Prod
* Visitor's country not keeping the label once saved (shows the value)
* style miss-aligned rules controls 

### Checklist
- [ ] Tests
- [ ] Translations
- [ ] Security Implications Contemplated (add notes if applicable)

### Additional Info
** any additional useful context or info **

### Screenshots
DOTCMS v3 (expected)
image

### CODE REVIEW

**Comment by alfredo-dotcms** [comment]
this is how it looks, after fixes
image

**Comment by alfredo-dotcms** [comment]
DEMO - current

image

File: libs/dot-rules/src/lib/components/dot-autocomplete-tags/dot-autocomplete-tags.component.css [hunk 1]
```diff
@@ -1,3 +1,11 @@
 .rules__engine-container p-autocomplete .ng-dirty.ng-valid input {
     border-left: none;
 }
+
+.cw-rule-actions :host ::ng-deep p-autocomplete span.p-autocomplete-multiple{
```
Comment (fmontes) [inline libs/dot-rules/src/lib/components/dot-autocomplete-tags/dot-autocomplete-tags.component.css:5]: What fixes this?

File: libs/dot-rules/src/lib/condition-types/serverside-condition/serverside-condition.ts [hunk 2]
```diff
@@ -172,15 +172,19 @@ export class ServersideCondition {
...
+
                 const input = this.getInputFor(paramDef.inputType.type, param, paramDef);
-                this._inputs.push(input);
+                this._inputs[paramDef.priority] = input;
+
             });
 
+            this._inputs = this._inputs.filter(i => i);
+
+            if (this._inputs.length <= 2 && change.componentInstance.currentValue instanceof ConditionModel) {
```
Comment (fmontes) [inline libs/dot-rules/src/lib/condition-types/serverside-condition/serverside-condition.ts:184]: Can you extract this condition to a function with a more understandable name?

File: libs/dotcms-js/src/lib/core/logger.service.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import { Injectable } from '@angular/core';
 import { StringUtils } from './string-utils.service';
-import { environment } from '../environments/environment';
+import { environment } from '../../../../../apps/dotcms-ui/src/environments/environment';
```
Comment (fmontes) [inline libs/dotcms-js/src/lib/core/logger.service.ts:3]: https://nx.dev/latest/react/guides/environment-variables

**Review by fmontes — CHANGES_REQUESTED** [review]

File: libs/dot-rules/src/lib/components/dropdown/dropdown.ts [hunk 4]
```diff
@@ -32,7 +32,6 @@ import { SelectItem } from 'primeng/api';
             [required]="minSelections > 0"
             [placeholder]="placeholder"
             [options]="dropdownOptions | async"
-            [editable]="allowAdditions"
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply libs/dot-rules/src/lib/components/dropdown/dropdown.ts:35]: `p-dropdown` is used in `cw-input-dropdown` only in dot-rules, and I looked through all places in rules and no one uses the editable option in `cw-input-dropdown`

File: libs/dot-rules/src/lib/services/Condition.ts [hunk 5]
```diff
@@ -39,7 +39,7 @@ export class ConditionService {
         return json;
     }
 
-    static fromServerConditionTransformFn(condition: ICondition): ConditionModel {
+    static fromServerConditionTransformFn(condition: ICondition, showLogs = false): ConditionModel {
```
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply libs/dot-rules/src/lib/services/Condition.ts:42]: at this moment and due to the task scope this is a workaround, a new task can be created later to implement a logger service that can be used through the whole app.

so @fmontes  do you suggest to keep this change or leave it as it was before?

**Review by hmoreras — APPROVED** [review]

File: libs/dot-rules/src/lib/services/Condition.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (alfredo-dotcms) [reply libs/dot-rules/src/lib/services/Condition.ts]: @fmontes Logger changes reverted!

**Review by fmontes — APPROVED** [review]

**Pull request merged**